### PR TITLE
fix https://github.com/Netflix/conductor/issues/1896

### PR DESCRIPTION
**Pull request by Yingsheng-eroad**

### CODE REVIEW

**Comment by Yingsheng-eroad** [comment]
@demichej @jvemugunta Can you review this PR? Thanks.

**Comment by Yingsheng-eroad** [comment]
This test pass successfully in my local. Don't know why it failed.

**Comment by apanicker-nflx** [comment]
Thank you for the contribution.

@rickfish @mactaggart @mashurex Could you please help with reviewing this PR? Thank you.

File: postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java [hunk 1]
```diff
@@ -90,6 +92,9 @@ private void flywayMigrate(DataSource dataSource) {
 
         flyway.setLocations(Paths.get("db","migration_postgres").toString());
         flyway.setDataSource(dataSource);
+        flyway.setBaselineOnMigrate(true);
+        flyway.setBaselineVersion(MigrationVersion.fromVersion("0"));
+        flyway.setTable("schema_version");
```
Comment (apanicker-nflx) [inline postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java:97]: Please refrain form hardcoding the table name here.

File: postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java [hunk 2]
```diff
@@ -90,6 +92,9 @@ private void flywayMigrate(DataSource dataSource) {
 
         flyway.setLocations(Paths.get("db","migration_postgres").toString());
         flyway.setDataSource(dataSource);
+        flyway.setBaselineOnMigrate(true);
```
Comment (apanicker-nflx) [inline postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java:95]: Consider abstracting this out into configuration.

**Comment by rickfish** [comment]
I don't see issues with adding those lines. I can't test MySQL but if those lines are added to the PostgresDataSourceProvider then they probably should be added to MySQLDataSourceProvider as well.

File: postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java (hunk 1 above)
Comment (mashurex) [reply postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java:97]: I agree, we use multiple Flyway migrations in the same schema and the version table must be customizable. I would add that `schema_version` is just a Flyway default, I have typically recommended to _always_ use something more descriptive like `${APP_NAME}_schema_version, i.e. `conductor_schema_version`.

**Comment by mashurex** [comment]
I have a general question on the scope of the commit. If this is around setting up Flyway to do baseline migrations (which I think is a fantastic idea), why are there incoming changes to DAO classes and tests?

File: postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java (hunk 1 above)
Comment (Yingsheng-eroad) [reply postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java]: > Please refrain form hardcoding the table name here.

Updated, remove the hardcoding name.

File: postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java (hunk 2 above)
Comment (Yingsheng-eroad) [reply postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java]: > Consider abstracting this out into configuration.

removed.

**Comment by apanicker-nflx** [comment]
Please rebase against the latest dev branch.

**Comment by Yingsheng-eroad** [comment]
> I don't see issues with adding those lines. I can't test MySQL but if those lines are added to the PostgresDataSourceProvider then they probably should be added to MySQLDataSourceProvider as well.

MySQL doesn't have this issue.

**Comment by Yingsheng-eroad** [comment]
> Please rebase against the latest dev branch.

Done.

**Comment by Yingsheng-eroad** [comment]
> I have a general question on the scope of the commit. If this is around setting up Flyway to do baseline migrations (which I think is a fantastic idea), why are there incoming changes to DAO classes and tests?
I had another PR to fix issue #1892. The PR was merged but shown in the latest dev.
https://github.com/Netflix/conductor/pull/1899/files

File: postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java [hunk 3]
```diff
@@ -13,9 +13,11 @@
 package com.netflix.conductor.postgres;
 
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.internal.cglib.proxy.$Factory;
```
Comment (apanicker-nflx) [inline postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java:16]: Is this import needed?

Comment (Yingsheng-eroad) [reply postgres-persistence/src/main/java/com/netflix/conductor/postgres/PostgresDataSourceProvider.java]: > Is this import needed?

Thanks. Removed. Don't even know how it got there.

**Comment by apanicker-nflx** [comment]
Please rebase against 2.31 branch and resolve conflicts.

**Comment by apanicker-nflx** [comment]
@Yingsheng-eroad is this still valid?

**Comment by apanicker-nflx** [comment]
Closing due to inactivity. Please reopen if you are still working on this.

**Pull request closed without merging**